sotrplib/sources/sources.py

```diff
@@ -1,4 +1,252 @@
+            prev_crossmatch_names=self.crossmatch_names,
+            prev_crossmatch_probabilities=self.crossmatch_probabilities,
+            prev_crossmatch_fluxes=self.crossmatch_fluxes,
+            prev_crossmatch_frequencies=self.crossmatch_frequencies,
+            new_crossmatch_names=new_names,
+            new_crossmatch_probabilities=new_probabilities,
+            new_crossmatch_fluxes=new_fluxes,
+            new_crossmatch_frequencies=new_frequencies,
+        )
+
+        if replace:
+            self.crossmatch_names = new_names
+            self.crossmatch_probabilities = new_probabilities
+            self.crossmatch_fluxes = new_fluxes
+            self.crossmatch_frequencies = new_frequencies
+        else:
+            for i, name in enumerate(new_names):
+                if name not in self.crossmatch_names:
+                    self.crossmatch_names.append(name)
+                    self.crossmatch_probabilities.append(new_probabilities[i])
+                    self.crossmatch_fluxes.append(new_fluxes[i])
+                    self.crossmatch_frequencies.append(new_frequencies[i])
+                else:
+                    self.crossmatch_probabilities[self.crossmatch_names.index(name)] = (
+                        new_probabilities[i]
+                    )
+                    self.crossmatch_fluxes[self.crossmatch_names.index(name)] = (
+                        new_fluxes[i]
+                    )
+                    self.crossmatch_frequencies[self.crossmatch_names.index(name)] = (
+                        new_frequencies[i]
+                    )
+        self.log.info("registeredsource.crossmatch_updated", RegisteredSource=self)
+        return
+
+
+class ForcedPhotometrySource(RegisteredSource):
+    """
+    A source object specifically for forced photometry measurements.
+
+    Since it's forced photometry, it is assumed to be a RegisteredSource which
+    undergoes a particular forced photometry flux measurement.
+
+    """
+
+    def __init__(
+        self,
+        ra: AstroPydanticQuantity[u.deg],
+        dec: AstroPydanticQuantity[u.deg],
+        flux: AstroPydanticQuantity[u.mJy] | None = None,
+        err_flux: AstroPydanticQuantity[u.mJy] | None = None,
+        source_id: str | None = None,
+        source_type: Literal["Extragalactic", "Star", "Asteroid", "Unknown", "Galaxy"]
+        | None = None,
+        crossmatch_names: list | None = None,
+        crossmatch_probabilities: list | None = None,
+        extended: bool | None = None,
+        err_ra: AstroPydanticQuantity[u.deg] | None = None,
+        err_dec: AstroPydanticQuantity[u.deg] | None = None,
+        fwhm_ra: AstroPydanticQuantity[u.deg] | None = None,
+        fwhm_dec: AstroPydanticQuantity[u.deg] | None = None,
+        fit_method: Literal["2D_Gaussian", "pixell.at", "other"] = "2D_Gaussian",
+        fit_params: dict | None = None,
+        log: FilteringBoundLogger | None = None,
+    ):
```

Comment:
You don't need to/want to write an init function for a pydantic model; they're created by pydantic itself.